src/v/compat/id_allocator_generator.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+
+#include "cluster/errc.h"
+#include "cluster/id_allocator.h"
+#include "cluster/types.h"
+#include "compat/generator.h"
+#include "model/timeout_clock.h"
+#include "random/generators.h"
+
+#include <chrono>
+#include <cstdint>
+#include <limits>
+#include <type_traits>
+
+namespace compat {
+
+template<>
+struct instance_generator<cluster::allocate_id_request> {
+    static cluster::allocate_id_request random() {
+        return cluster::allocate_id_request{model::timeout_clock::duration(
+          random_generators::get_int<int64_t>(-100000, 100000) * 1000000)};
+    }
+
+    static std::vector<cluster::allocate_id_request> limits() {
+        return {
+          cluster::allocate_id_request{model::timeout_clock::duration(0)},
+          cluster::allocate_id_request{std::chrono::milliseconds::min()},
+          cluster::allocate_id_request{std::chrono::milliseconds::max()},
+        };
+    }
+};
+
+template<>
+struct instance_generator<cluster::allocate_id_reply> {
+    using errc_type = typename std::underlying_type_t<cluster::errc>;
+    static constexpr auto int64_min = std::numeric_limits<int64_t>::min();
+    static constexpr auto int64_max = std::numeric_limits<int64_t>::min();
+    static constexpr auto errc_min = static_cast<errc_type>(
```

Comment:
In my branch i think i've done something like:
```
template<>
struct instance_generator<cluster::errc>
```
and then having `random()` return every type wrapped in a `random_choice` call. Feel free to take that from me, then i can cherry-pick it from your work